fastlane/lib/fastlane/actions/update_code_signing_settings.rb

```diff
@@ -16,84 +16,86 @@ def self.run(params)
+                UI.important("Skipping #{config.name} not selected (#{params[:build_configurations].join(',')})")
+                next
               end
             end
-            UI.important("Set Code Sign identity to: #{params[:code_sign_identity]} for target: #{found_target[:name]}")
-          end
-          if params[:profile_name]
-            build_configuration_list.set_setting("PROVISIONING_PROFILE_SPECIFIER", params[:profile_name])
-            UI.important("Set Provisioning Profile name to: #{params[:profile_name]} for target: #{found_target[:name]}")
-          end
-          # Since Xcode 8, this is no longer needed, you simply use PROVISIONING_PROFILE_SPECIFIER
-          if params[:profile_uuid]
-            build_configuration_list.set_setting("PROVISIONING_PROFILE", params[:profile_uuid])
-            UI.important("Set Provisioning Profile UUID to: #{params[:profile_uuid]} for target: #{found_target[:name]}")
-          end
-          if params[:bundle_identifier]
-            build_configuration_list.set_setting("PRODUCT_BUNDLE_IDENTIFIER", params[:bundle_identifier])
-            UI.important("Set Bundle identifier to: #{params[:bundle_identifier]} for target: #{found_target[:name]}")
+
+            style_value = params[:use_automatic_signing] ? 'Automatic' : 'Manual'
+            set_build_setting(config, "CODE_SIGN_STYLE", style_value)
+
+            if params[:team_id]
+              set_build_setting(config, "DEVELOPMENT_TEAM", params[:team_id])
+              UI.important("Set Team id to: #{params[:team_id]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+            if params[:code_sign_identity]
+              set_build_setting(config, "CODE_SIGN_IDENTITY", params[:code_sign_identity])
+              UI.important("Set Code Sign identity to: #{params[:code_sign_identity]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+            if params[:profile_name]
+              set_build_setting(config, "PROVISIONING_PROFILE_SPECIFIER", params[:profile_name])
+              UI.important("Set Provisioning Profile name to: #{params[:profile_name]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+            # Since Xcode 8, this is no longer needed, you simply use PROVISIONING_PROFILE_SPECIFIER
+            if params[:profile_uuid]
+              set_build_setting(config, "PROVISIONING_PROFILE", params[:profile_uuid])
+              UI.important("Set Provisioning Profile UUID to: #{params[:profile_uuid]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+            if params[:bundle_identifier]
+              set_build_setting(config, "PRODUCT_BUNDLE_IDENTIFIER", params[:bundle_identifier])
+              UI.important("Set Bundle identifier to: #{params[:bundle_identifier]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+
```

Comment:
It's a little hard to follow what's going on with a long block inside of nested loop.

Suggestion: Collect the appropriate build configurations. Then map over them with a function that a function that takes a `config` and `params` and `target`(?). The return values can be used to accumulate `changed_targets` and `changed_build_configurations`.

Another option would be separate the collecting of the build configs as a separate function.